src/classes/Recipe.js

```diff
@@ -0,0 +1,35 @@
+class Recipe {
+    constructor(recipeDetails) {
+        this.id = recipeDetails.id;
+        this.name = recipeDetails.name;
+        this.image = recipeDetails.image;
+        this.portions = recipeDetails.portions;
+        this.instructions = recipeDetails.instructions;
+        this.tags = recipeDetails.tags;
+
+    }
+    showIngredientsNeeded() {
+        return this.portions.map((portion) => {
+            return portion.name;  
+        });
+    }
+    calcTotalIngredientCost() {
+        const ingredientsCost = this.portions.map((portion) => {
+            return (portion.amount * portion.cost).toFixed(2) / 100;
+        });
+        return ingredientsCost;
+    }
+    calcRecipeCost() {
```

Comment:
I humbly submit `calculateTotalRecipeCost` as a rename for this function

Reply:
Changed over, funnily this was my original naming convention